src/Database/Type/DateTimeType.php

```diff
@@ -103,7 +103,7 @@ public function toDatabase($value, Driver $driver)
      */
     public function toPHP($value, Driver $driver)
     {
-        if ($value === null || (int)$value === 0) {
+        if ($value === null || strpos($value, '-') !== false && (int)$value === 0) {
```

Comment:
new idea, that will save one function call: `strpos($value, '0000-00-00') === 0`